src/Microsoft.Azure.WebJobs.Host/Tables/AsyncCollectorArgumentBindingProvider.cs

```diff
@@ -63,14 +63,13 @@ public FileAccess Access
 
             public Type ValueType
             {
-                get { return typeof(ICollector<TElement>); }
+                get { return typeof(IAsyncCollector<TElement>); }
```

Comment:
These are very old copy/paste bugs that I discovered while debugging. The Debug Assert in RuntimeValueProvider was firing.